rfc/text/advanced/ap_authentication_scram.md

```diff
@@ -0,0 +1,387 @@
+
+{align="left"}
+```javascript
+    [4, "wamp-scram",
+        {
+            "nonce": "egVDf3DMJh0=SBmkFIh7sSo=",
+            "salt": "aBc+fx0NAVA=",
+            "kdf": "pbkdf2",
+            "iterations": 4096,
+            "memory": null
+        }
+    ]
+```
+
+where:
+
+1. `nonce|string`: A server-generatated nonce that is appended to the client-generated nonce sent in the previous `HELLO` message. See [Nonces](#nonces).
+2. `salt|string`: The base64-encoded salt for this user, to be passed to the key derivation function. This value is stored with each user record in the authentication database. See [Salts](#salts).
+3. `kdf`: The key derivation function (KDF) used to hash the password. This value is stored with each user record in the authentication database. See [Key Derivation Functions](#key-derivation-functions).
+4. `iterations|integer`: The execution time cost factor to use for generating the `SaltedPassword` hash. This value is stored with each user record in the authentication database.
+5. `memory|integer`: The memory cost factor to use for generating the `SaltedPassword` hash. This is only used by the Argon2 key derivation function, where it is stored with each user record in the authentication database.
+
+The client MUST respond with an ABORT message if `CHALLENGE.Details.nonce` does not begin with the client nonce sent in `HELLO.Details.nonce`.
+
+The client SHOULD respond with an ABORT message if it detects that the cost parameters are unusually low. Such low-cost parameters could be the result of a rogue server attempting to obtain a weak password hash that can be easily cracked. What constitutes unusually low parameters is implementation-specific and is not covered by this document.
+
+
+###### Final Client Authentication Message
+
+Upon receiving a valid `CHALLENGE` message, the client SHALL respond with an `AUTHENTICATE` message:
+
+{align="left"}
+```javascript
+    [5, "dHzbZapWIk4jUhN+Ute9ytag9zjfMHgsqmmiz7AndVQ=",
+        {
+            "nonce": "egVDf3DMJh0=SBmkFIh7sSo=",
+            "channel_binding": null,
+            "cbind_data": null
+        }
+    ]
+```
+where:
+
+1. `Signature|string` argument: The base64-encoded `ClientProof`, computed as described in the [SCRAM-Algorithms](#scram-algorithms) section.
+2. `nonce|string`: The concatenated client-server nonce from the previous `CHALLENGE` message.
+3. `channel_binding|string`: Optional string containing the channel binding type that was sent in the original `HELLO` message.
+4. `cbind_data|string`: Optional base64-encoded channel binding data. MUST be present if and only if `channel_binding` is not `null`. The format of the binding data is dependent on the binding type. For `tls-server-end-point`, this is a hash of the TLS server's certificate, computed as described in [RFC5929, section 4.1](https://tools.ietf.org/html/rfc5929#section-4.1).
```

Comment:
no, this isn't sth required for `tls-unqiue`, and that is the channel binding I would like to use ..